disaggregated_memory/memories/external_remote_memory_v2.py

```diff
@@ -0,0 +1,215 @@
+# Copyright (c) 2023-24 The Regents of the University of California
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met: redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer;
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""We need a class that extends the outgoing bridge from gem5. The goal
+of this class to have a MemInterface like class in the future, where we'll
+append mem_ranges within this interface."""
+
+from typing import (
+    List,
+    Sequence,
+    Tuple,
+)
+
+import m5
+from m5.objects import (
+    AddrRange,
+    ExternalMemory,
+    MemCtrl,
+    Port,
+    Tick,
+)
+from m5.util import (
+    fatal,
+    warn,
+)
+
+from gem5.components.boards.abstract_board import AbstractBoard
+from gem5.components.memory.memory import AbstractMemorySystem
+from gem5.utils.override import overrides
+
+
+class ExternalRemoteMemoryV2(AbstractMemorySystem):
```

Comment:
Don't call this V2

Reply:
That makes sense. I was thinking about it too. Here is the naming convention that I am thinking of.
`ExternalMemory` -> `AbstractMem`
`ExternalRemoteMemory` -> `AbstractMemorySystem`

Reply:
Done!